frontend/src/container/PlannedDowntime/PlannedDowntime.tsx

```diff
@@ -128,17 +134,19 @@ export function PlannedDowntime(): JSX.Element {
 					setEditMode={setEditMode}
 					searchValue={searchValue}
 				/>
-				<PlannedDowntimeForm
-					alertOptions={alertOptions || []}
-					initialValues={initialValues}
-					isError={isError}
-					isLoading={isLoading}
-					isOpen={isOpen}
-					setIsOpen={setIsOpen}
-					refetchAllSchedules={refetchAllSchedules}
-					isEditMode={isEditMode}
-					form={form}
-				/>
+				{isOpen && (
```

Comment:
why is this needed ?

Reply:
As the component and logic added under this is dependent on this flag only, and few state settings and logic I wanted to segregate all those under a separate file.